src/other/index.jl

```diff
@@ -214,8 +214,35 @@ end
 @inline Base.getindex(x::AbstractIndex, idx::AbstractRange{<:Integer}) =
     getindex(x, collect(Int, idx))
 @inline Base.getindex(x::AbstractIndex, ::Colon) = Base.OneTo(length(x))
-@inline Base.getindex(x::AbstractIndex, notidx::Not) =
-    setdiff(1:length(x), getindex(x, notidx.skip))
+@inline function Base.getindex(x::AbstractIndex, notidx::Not)
+    if notidx.skip isa AbstractVector 
+        toskip = filter(t -> haskey(x, t), notidx.skip)
+        return setdiff(1:length(x), toskip)
```

Comment:
Maybe I'm missing something, but can't you omit the `filter` here and just use `setdiff`? E.g., this works:

```julia
julia> setdiff(1:3, 3:4)
2-element Array{Int64,1}:
 1
 2
```

Also, what happens if you have a vector of symbols, e.g. `Not([:c, :d])`?

Reply:
right - `toskip` should be probably be a comprehension calling `getindex` with `haskey` after the `if` part.